.gitlab-ci.yml

```diff
@@ -12,5 +17,107 @@ include:
       - '.gitlab-ci-check-commits.yml'
       - '.gitlab-ci-check-license.yml'
 
+test:
+  stage: test
+  image: golang:1.17
+  before_script:
+    - apt-get update && apt-get install -yyq libssl-dev libglib2.0-dev dbus clang-format-9
+    - GO111MODULE=off go get -u github.com/jstemmer/go-junit-report
+  script:
+    - git ls-tree -r --name-only HEAD | grep -v vendor/ | grep '\.[ch]$' | xargs clang-format-9 -i
+    - make coverage
+    - make
+    - mkdir -p tests/unit-coverage && find . -name 'coverage.txt' -exec cp --parents {} ./tests/unit-coverage \;
+    - tar -cvf $CI_PROJECT_DIR/unit-coverage.tar tests/unit-coverage
+  tags:
+    - mender-qa-slave
+  artifacts:
+    expire_in: 2w
+    reports:
+      junit: report.xml
+    paths:
+      - unit-coverage.tar
+
+publish:tests:
+  stage: publish
+  image: golang:1.17-alpine3.14
+  dependencies:
+    - test
+  before_script:
+    # Install dependencies
+    - apk add --no-cache git
+    - GO111MODULE=off go get -u github.com/mattn/goveralls
+    # Coveralls env variables:
+    #  According to https://docs.coveralls.io/supported-ci-services
+    #  we should set CI_NAME, CI_BUILD_NUMBER, etc. But according
+    #  to goveralls source code (https://github.com/mattn/goveralls)
+    #  many of these are not supported. Set CI_BRANCH,
+    #  and pass few others as command line arguments.
+    #  See also https://docs.coveralls.io/api-reference
+    - export CI_BRANCH=${CI_COMMIT_BRANCH}
+  script:
+    - tar -xvf unit-coverage.tar
+    - goveralls
+      -repotoken ${COVERALLS_TOKEN}
+      -service gitlab-ci
+      -jobid $(git rev-parse HEAD)
+      -covermode set
+      -flagname unittests
+      -parallel
+      -coverprofile $(find tests/unit-coverage -name 'coverage.txt' | tr '\n' ',' | sed 's/,$//')
+
+generate-qa-trigger:
```

Comment:
A bit unsure about the trigger. I think we can live without it, we will still have the nightly builds with latest master of this tool, and for release branches it might be an overkill for this tool.

We will also need to revisit the release process for independent components, and then these triggers would be anyway deprecated.